### ISSUE TITLE
TWE: HfLA Volunteer Research Plans and Goals Chart: Create

### ISSUE THREAD

**Issue by pandanista** [hackforla/internship#374]
### Overview
We need to update the Research Plans and Goals chart on Wiki with the research plan(s) info so we could provide an overview for external use.

During these tasks you will be asked to add link to the Resources/Instructions section, and that section is at the bottom of this issue (go view it now).

#### Customize Resource Links
- Customize Resource for the Research Plan's Google Drive Folder
    - [x] Go to "Google Drive Folder: Research by Participant Type" (Resources # 1.01)
    - [x] Choose "HfLA Volunteers" folder and right-click on the folder
    - [x] Choose "Get Link" 
    - [x] Choose "Copy link" and "Done"
    - [x]  Choose "Edit" under the three dots to access the GitHub Write mode
    - [x] Update Resource # 2.01 with the link you just copied. Place it into parentheses at the end of the line.
    - [x] Choose "Update comment" in Github and make sure all the checkboxes above have been checked <img width="100" alt="Screenshot 2023-02-21 at 6 47 15 PM" src="https://user-images.githubusercontent.com/57029070/220508425-5d0c0a1b-ef87-4ef2-a683-d4a21d5b2672.png">
    - [x] Identify if there is any research plan document in the folder (Resource # 2.01)
        - [x] If the research plan(s) exists,  
            1. Right-click on each research plan Google Doc 
            1. Choose "Rename" and copy the texts       
            1. Paste the texts in brackets in Resources as # 2.01.01, 2.01.02, 2.01.03, etc 
            1. Again, right-click on the research plan Google Doc in Google Drive
            1. Choose "Get Link" 
            1. Choose "Copy link" and "Done"
            1. Update Resources # 2.01.01,  2.01.02, 2.01.03, etc., with the link you just copied. Place it into parentheses at the end of the line.
        - If any research plan(s) is missing, 
            - [ ] Choose the gear in the Labels section <img width="500" alt="Screenshot 2023-04-19 at 8 06 27 PM" src="https://user-images.githubusercontent.com/57029070/233248009-99d8d10c-112a-4bc9-8fd5-b1e0098aea23.png">
            - [ ] Put on the dependency label
            - [ ] Copy
            ```
            ### Dependencies
            - [ ] The research plan(s) needs to be created. See issue # 
            ```
            - [ ] Choose "Edit" under the three dots to access the GitHub Write mode
            - [ ] Paste what you just copied to the beginning of the issue above the `Overview` section
            - [ ] Choose `Update comment` <img width="100" alt="Screenshot 2023-02-21 at 6 47 15 PM" src="https://user-images.githubusercontent.com/57029070/220508425-5d0c0a1b-ef87-4ef2-a683-d4a21d5b2672.png">
            - [ ] Choose `Issues` located on the top of the GitHub page to open a new tab in your browser <img width="600" alt="Screenshot 2023-06-15 at 5 18 07 PM" src="https://github.com/hackforla/internship/assets/57029070/bcee2deb-8032-4ea3-863f-dd7e5b3bb898">
            - [ ] In the search bar, include the missing research plan number, i.e. RP001, to do a search to see if the research plan has an existing issue. Make sure to check both open and closed issue. <img width="700" alt="Screenshot 2023-06-15 at 4 18 57 PM" src="https://github.com/hackforla/internship/assets/57029070/7ab3bec1-1fa4-4fc3-a801-362785686d53">
            - [ ] If there is an existing issue for creating that research plan, add the issue number after "See issue # " in the dependencies section so we can track  the status of the research plan issue
                - [ ] Assign the existing research plan issue to yourself if no one is assigned so you can start working on that since it is a dependency
            - [ ] If there is non existing issue for creating that research plan, create a new issue using [research plan template](https://github.com/hackforla/internship/issues/new?assignees=&labels=feature%3A+board+%2F+google+drive+%2F+tools%2C+feature%3A+missing%2C+feature%3A+research+plan%2C+milestone%3A+missing%2C+role%3A+UI%2FUX+research%2C+size%3A+1pt&projects=&template=research--research-plan.md&title=TWE%3A+Create+Research+Plan+for+RP%5B%23%23%23%5D)
                - [ ] Customize the new issue using the missing research plan's number, RP###
                - [ ] Once the new issue is created, include the issue number after "See issue # " in the dependencies section
                - [ ] Assign the new issue to yourself so you can start working on the new issue since it is a dependency
            - [ ] Make a comment in this issue as an update to inform the team that there is a missing research plan and include the issue number for creating the research plan
            - [ ] Put this issue back to the icebox
            - [ ] Remove yourself from this issue
        - [x] Choose "Update comment" in Github and make sure all the checkboxes above have been checked 
- Customize Resource for Wiki Page Link
    - [x] Go to "Research Documents by Participant Type" section on the Research Output Overview Wiki Page (Resources # 1.02)
    - [x] Choose `HfLA Volunteers` so it directs to the `HfLA Volunteers Research Overview` Wiki page
    - [x] Copy the URL of `HfLA Volunteers Research Overview` Wiki page in the browser
    - [x] Update Resource # 2.02 with the link you just copied. Place it into parentheses at the end of the line.
    - [x] Choose "Update comment" in Github and make sure all the checkboxes above have been checked 
- Customize Resource for Research Plans and Goals Wiki Page
    - [x] Select the hyperlink in the `HfLA Volunteers Research Plans and Goals (public page)` section on `HfLA Volunteers Research Overview` page so it directs to `HfLA Volunteers Research Plans and Goals` Wiki page
    - [x] Copy the URL of `HfLA Volunteers Research Plans & Goals` Wiki page in the browser
    - [x] Update Resource # 2.03 with the URL you just copied. Place it into parentheses at the end of the line.
    - [x] Choose "Update comment" in Github and make sure all the checkboxes above have been checked 
- Customize the spreadsheet for generating the Research Plans and Goals chart
    - [x] Open the template "TWE: Template: Research Plans and Goals Chart Spreadsheet" (Resources # 1.03)
    - [x] Choose the "Use Template" button to save it to your own drive
    - Rename the file by following the steps:
        - [x] Copy 
        ```
        TWE: IS22: HfLA Volunteers Research Plans and Goals Chart Spreadsheet
        ```
        - [x] Choose "File" > "Rename", and paste the texts you copied into the top field    
       <img width="200" alt="Screenshot 2023-01-30 at 1 35 41 PM" src="https://user-images.githubusercontent.com/57029070/215600780-d8004229-17a3-409a-9d91-e4af9c904824.png"> <img width="400" alt="Screenshot 2023-02-27 at 2 51 11 PM" src="https://user-images.githubusercontent.com/57029070/221705684-e600627e-cbff-4642-ae62-c6db9dd7ea28.png">
    - [x] Move the file into the Internship's shared drive in this Research Plan's Google Drive folder
        1. Choose File > Move (or Move icon)
        4. Choose the back arrow to access all your drives 
        5. Choose Shared drives > Internships > Internships > Research > Research by Participant Type > HfLA Volunteers
        6. Choose "Move here"
        7. Choose "Move" when it asks if you want to change ownership to a shared drive
        8. Dismiss the notification that says the file is now in the shared drive
    - [x] Update Resource # 2.04 link
        1. In the Research Plans and Goals Chart Spreadsheet, choose "Share" (upper right corner)
        3. Choose "Copy link" and "Done"
        4. Update Resource # 2.04 with the link you just copied. Place it into parentheses at the end of the line.
        5. Choose "Update comment" in Github and make sure all the checkboxes above have been checked 

#### Customize the Research Plans and Goals Chart Spreadsheet
- [x] Open the Research Plans and Goals Chart Spreadsheet in Resources # 2.04 if it is not open
- In "All with edit links" tab
    - [x] Fill out Columns A - F based on research plan and roadmap doc, starting from Row 2 (Resources # 2.02 and # 2.01)
    - [x] Add the links to the research plans in Column G
        1. Copy the matching research plan's URL from Resources 2.01's sub-section, such as 2.01.01
        1. In Column G, choose the cell that matches the research plan
        1. Click on the link icon <img width="30" alt="Screenshot 2023-02-27 at 3 31 50 PM" src="https://user-images.githubusercontent.com/57029070/221715925-7804635a-49be-45e3-9d81-06c69056f29e.png">
        1. Paste the URL you copied into the text box. Update the URL by removing the `/edit?usp=share_link` string and add `/preview` at the end of the URL so all participants can view the document <img width="200" alt="Screenshot 2023-02-27 at 3 32 18 PM" src="https://user-images.githubusercontent.com/57029070/221716041-a0d6450e-99b6-442d-a086-39083bc9ecb1.png">
        1. Choose `Apply`
    - [x] Add the link to the Google Form survey or script(s) if they exist in Column H and Column I
        1. Choose the folder linked in Resources # 2.01
        1. Open each research plan folder
        1. Choose the Google Form (could be a survey or an interview script depending on the research method used for the study) in the research plan folder
        1. Choose "Send" <img width="50" alt="Screenshot 2023-02-27 at 6 07 11 PM" src="https://user-images.githubusercontent.com/57029070/221736079-1cfb9b2a-0bcd-4a1b-a6c7-0e3e868d6ee6.png">
        1. Choose the link icon <img width="50" alt="Screenshot 2023-02-27 at 6 07 31 PM" src="https://user-images.githubusercontent.com/57029070/221735981-a43f935f-7b61-475a-84c7-52803e3bf184.png"> and choose "Copy"
        1. In Column H, choose the cell that corresponds with the research plan 
        1. Click on the link icon <img width="40" alt="Screenshot 2023-02-27 at 3 31 50 PM" src="https://user-images.githubusercontent.com/57029070/221715925-7804635a-49be-45e3-9d81-06c69056f29e.png">
        1. Paste the URL you copied into the text box <img width="200" alt="Screenshot 2023-02-27 at 3 32 18 PM" src="https://user-images.githubusercontent.com/57029070/221716041-a0d6450e-99b6-442d-a086-39083bc9ecb1.png">
        1. Choose `Apply` 
        1. Copy the URL from the browser tab in which the Google Form is open
        1. In Column I, choose the cell that corresponds with the research plan 
        1. Click on the link icon <img width="40" alt="Screenshot 2023-02-27 at 3 31 50 PM" src="https://user-images.githubusercontent.com/57029070/221715925-7804635a-49be-45e3-9d81-06c69056f29e.png">
        1. Paste the URL you copied from the browser into the text box <img width="200" alt="Screenshot 2023-02-27 at 3 32 18 PM" src="https://user-images.githubusercontent.com/57029070/221716041-a0d6450e-99b6-442d-a086-39083bc9ecb1.png">
        1. Choose `Apply` 
    - [ ] If the Google Form survey/script (Column H) does not exist in research plan folders 
        1. Make a comment in this issue as an update to let the team know the script(s) needs to be created and added in the Research Plan and Goals chart once it is available
        1. Check in with the team at the next research meeting to get an estimate on when the Google Form will be ready
- [x] If any of the research plans use interviews as the research method, update the "Interviews published version" tab 
    1. In "All with edit links" tab, choose the filter in Research Method(s) (Column C) <img width="80" alt="Screenshot 2023-02-27 at 7 08 02 PM" src="https://user-images.githubusercontent.com/57029070/221744300-2d3d0016-7ab7-4d22-a78c-89c20a26e8bb.png">
    1. Check "Interview" and uncheck the rest of methods <img width="80" alt="Screenshot 2023-02-27 at 7 08 39 PM" src="https://user-images.githubusercontent.com/57029070/221744485-2296af85-e2c2-41c1-8583-39e1ca1f4e3e.png">
    1. Choose "OK" <img width="50" alt="Screenshot 2023-02-27 at 7 09 11 PM" src="https://user-images.githubusercontent.com/57029070/221744604-e6f22e96-b056-4d32-b88a-0aea63b36b62.png">
    1. The research plans using interviews as the research method are populated
    1. Copy the content (texts and URLs) from Columns A-B and Columns D-H with no header from Row 2
    1. Paste the content into the "Interviews published version" tab starting from Row 3
- [x] If interviews are not used, leave the "Interviews published version" tab blank
- [ ] If any of the research plans use surveys as the research method, update the "Surveys published version" tab 
    1. In "All with edit links" tab, choose the filter in Research Method(s) (Column C) <img width="80" alt="Screenshot 2023-02-27 at 7 08 02 PM" src="https://user-images.githubusercontent.com/57029070/221744300-2d3d0016-7ab7-4d22-a78c-89c20a26e8bb.png">
    1. Check "Survey" and uncheck the rest of methods <img width="80" alt="Screenshot 2023-02-27 at 7 08 49 PM" src="https://user-images.githubusercontent.com/57029070/221745301-b75472dc-c360-4ba9-9efe-fef00cfeb54f.png">
    1. Choose "OK" <img width="50" alt="Screenshot 2023-02-27 at 7 09 11 PM" src="https://user-images.githubusercontent.com/57029070/221744604-e6f22e96-b056-4d32-b88a-0aea63b36b62.png">
    1. The research plans using surveys as the research method are populated
    1. Copy the content (texts and URLs) from Columns A-B and Columns D-H with no header from Row 2
    1. Paste the content into the "Surveys published version" tab starting from Row 3
- [x] If surveys are not used, leave the "Surveys published version" tab blank
- [ ] If there is any other research method to be used or used, choose "[research method] published version" tab 
    - [ ] Click on the green down arrow <img width="150" alt="Screenshot 2023-02-27 at 3 11 35 PM" src="https://user-images.githubusercontent.com/57029070/221710452-3b80bbe3-1545-4a09-8278-03903720ce96.png"> and choose `Rename` <img width="50" alt="Screenshot 2023-02-27 at 3 16 41 PM" src="https://user-images.githubusercontent.com/57029070/221711309-314ee1e9-4864-41fe-a01f-a4160b666bb0.png">
    - [ ] Update `[research method]` in the name of the tab with the method to be used or used, such as "contextual inquiry", etc.
    - [ ] Update row 2 `[research method]` with the method to be used or used, such as "contextual inquiry", etc.
    - [ ] Populate the research plans using other research method(s) one by one
        1. In "All with edit links" tab, choose the filter in Research Method(s) (Column C) <img width="80" alt="Screenshot 2023-02-27 at 7 08 02 PM" src="https://user-images.githubusercontent.com/57029070/221744300-2d3d0016-7ab7-4d22-a78c-89c20a26e8bb.png">
        1. Check one research method other than interview and survey, and uncheck all other methods 
        1. Choose "OK" <img width="50" alt="Screenshot 2023-02-27 at 7 09 11 PM" src="https://user-images.githubusercontent.com/57029070/221744604-e6f22e96-b056-4d32-b88a-0aea63b36b62.png">
        1. The research plans using that particular research method are populated
        1. Copy the content (texts and URLs) from Columns A-B and Columns D-H with no header from Row 2
        1. Paste the content into the other research method's "published version" tab from Row 3
        1. Make one tab for each of the other research methods other than interviews and surveys if needed
- [x] If no other research method(s) is used, leave the "[research method] published version" tab blank
- [x] Update the "Published plans & goals by research methods" tab
    1. Copy the entire spreadsheet starting from Row 2 (no header) in "Interviews published version" tab if interviews are used as a research method. Otherwise, skip the "Interviews published version" spreadsheet.
    1. Paste the spreadsheet into the "Published plans & goals by research methods" tab
    1. Copy the entire spreadsheet starting from Row 2 (no header) in "Surveys published version" tab if surveys are used as a research method. Otherwise, skip the "Surveys published version" spreadsheet.
    1. Paste the spreadsheet into the "Published plans & goals by research methods" tab
    1. Copy the entire spreadsheet starting from Row 2 (no header) in "[research method] published version" tab if any other methods are used. Otherwise, skip the "[research method] published version" spreadsheet.
    1. Paste the spreadsheet into the "Published plans & goals by research methods" tab

#### Update Research Plan and Goals Wiki page
- Convert the spreadsheet into Markdown language
- [x] Copy the spreadsheet (with header) in the "Published plans & goals by research methods" tab in Resource # 2.04
- [x] Open the link in Resources # 1.04
- [x] Paste the spreadsheet you just copied into the box with "Paste Table Here! " being displayed <img width="200" alt="Screenshot 2023-02-27 at 8 04 53 PM" src="https://user-images.githubusercontent.com/57029070/221753856-06d40e21-ee9b-42b9-a464-9c9f978dbbda.png">
- [x] Choose "Submit" <img width="50" alt="Screenshot 2023-02-27 at 8 05 47 PM" src="https://user-images.githubusercontent.com/57029070/221751672-af7d145f-5c49-4a5e-a245-3d920b6e1857.png">
- [x] Use the default settings once the table is converted to Markdown. By default, "Outer pipes" and "Cell padding" are checked, and "No sorting" is the option selected on the drop down menu <img width="300" alt="Screenshot 2023-02-27 at 8 09 08 PM" src="https://user-images.githubusercontent.com/57029070/221751616-a8453e0f-ce63-446f-88c2-5b25bd55d60f.png">
- [x] Copy the converted Markdown table from the box
- Update Wiki page
- [x] Open the link in Resources # 2.03
- [x] Choose "Edit"
- [x] Paste the converted Markdown table into the text box under the `### Mentor Research Plans & Goals` section
- [x] Remove outdated Research Plans & Goals table if there was any
- [x] Choose "Save page"

####
- [x] Review with UX Lead
- [x] Product sign-off

#### Resources for creating this issue
1.01 [Google Drive Folder: Research by Participant Type](https://drive.google.com/drive/folders/1f5Qgq-ikT_UwcgRBuoBamqY0Wacsg9f5?usp=share_link)
1.02 [Research Output Overview](https://github.com/hackforla/internship/wiki/Research-Output-Overview)
1.02.01 Examples of the Research Plans and Goals Charts: [Mentor Research Plans and Goals](https://github.com/hackforla/internship/wiki/Mentor-Research-Plans-and-Goals); [Intern-Research-Plans-and-Goals](https://github.com/hackforla/internship/wiki/Intern-Research-Plans-and-Goals)
1.02.02 Example of the spreadsheet prior to the Github Wiki chart: [Intern Research List](https://docs.google.com/spreadsheets/d/1Y2VeHEUhmx4UX6y8CtbCHy2ilH76lWO4_xAVoqnLyME/edit#gid=1271462147)
1.03 [TWE: Template: Research Plans and Goals Chart](https://docs.google.com/spreadsheets/d/1TA0AGHKFhb4b3IbylfGAytoHmaDmfjsSuPPuEVlqCF0/template/preview)
1.04 [Convert spreadsheet to Markdown language](https://tabletomarkdown.com/convert-spreadsheet-to-markdown/)

#### Resources gathered during the completion of this issue
2.01 [Google Drive Folder: HfLA Volunteers](https://drive.google.com/drive/folders/10hqcOE54U71UZd_Vtu4WPsXSopXZnDdk?usp=share_link)
2.01.01 [IP: Research Plan to find transferable skills](https://docs.google.com/document/d/15jkt2K0k3amrMMCHopNwurRrZpFyXDFhsjZ1YZ0qP9o/edit)
2.01.02 [TWE: Research Plan: RP020 Intern Interaction with Hack for LA Website Team](https://docs.google.com/document/d/1GK3jaUlJerLtUFZhIXXAM2oDZDeYBWyooPXH0Z3IFhw/edit)
2.02 [HfLA Volunteers Research Overview](https://github.com/hackforla/internship/wiki/HfLA-Volunteers-Research-Overview)
2.03 [HfLA Volunteers Research Plans and Goals Chart](https://github.com/hackforla/internship/wiki/HfLA-Volunteers-Research-Plans-and-Goals)
2.04 [HfLA Volunteers Research Plans and Goals Chart Spreadsheet](https://docs.google.com/spreadsheets/d/1bxQdIEqgOOFhW6Ev56tROQy4RLJ9aeFG9kg465jHJOs/edit?usp=sharing)

**Comment by pandanista** [comment]
At the last meeting on 2/28, I mentioned the issue was created based on a template we are still testing—meaning, while working on this issue, you might run into instructions that are not clear enough, or confusing, or direct you to a wrong folder, etc. Therefore, please leave any feedback you have below as comments as you check off the action items. Much appreciated. :) 

Please see below for steps and examples for providing feedback:
### Action Items for Providing Feedback on Testing Templates
- [ ] Provide feedback for this issue
     - [ ] Add feedback in the form of (1) questions and/or (2) explicit instructions for how the content should be improved. Feedback in a Google Doc or spreadsheet should be given in the form of document comments. 
     - [ ] Determine the accuracy of the content for each section (action items listed in the issue above, Google doc, or Google sheet). If you find an inaccuracy, error, or mis-categorization, explain how the information presented could be improved.
     - [ ] For each suggested improvement, provide a link, reference, or a screenshot within your comment to justify your suggestion.
        - Examples:  
        <img width="700" alt="Screenshot 2023-03-06 at 6 37 40 PM" src="https://user-images.githubusercontent.com/57029070/223306823-2a79b9df-c996-4ac7-b1b0-e8aeaf0e9170.png">
        <img width="700" alt="Screenshot 2023-03-06 at 6 37 10 PM" src="https://user-images.githubusercontent.com/57029070/223306826-0d29a792-c4fa-4458-96ab-3614b0b66a76.png">
        <img width="700" alt="Screenshot 2023-03-06 at 6 36 56 PM" src="https://user-images.githubusercontent.com/57029070/223306827-ee6c9cad-9fd4-476f-9e8f-bde64a2da151.png">
        <img width="700" alt="Screenshot 2023-03-06 at 6 36 46 PM" src="https://user-images.githubusercontent.com/57029070/223306831-4105863d-192d-4860-810f-efb17cc15e44.png">
- [ ] Once all the steps have been completed, reach out to the current UX Lead to determine if they can incorporate the feedback in the issue template.

**Comment by pandanista** [comment]
Hi Drew, removing you from this issue for now since we are doing the research plan audits.

**Comment by AndrewSalvatore** [comment]
Progress: done except last blockers

Blockers: In the Research Plans and Goals Chart excel sheet ("All with edit links" tab), what goes in the "Day" column? Also, if the script is in the RP doc, is it correct to just link the RP doc in column H?

Availability: 6 hours

ETA: today after blockers are unblocked

Pictures or links (if necessary):none

**Comment by AndrewSalvatore** [comment]
Progress: completed, waiting for UX lead sign-off

Blockers: none

Availability: 6 hours

ETA: done

Pictures or links (if necessary): none

**Comment by AndrewSalvatore** [comment]
**Template Action Items to change/improve:**
- "Identify if any of the research plans are created in the folder"
- I was confused by this wording. Perhaps change this to "Identify if there any research plans in the folder"

<img width="626" alt="image" src="https://user-images.githubusercontent.com/122848208/231630475-85c88364-aa4a-4f25-9e4b-2ade052c445c.png">
- There needs to be a differentiation between the Research Plans and Goals chart on the Wiki and the spreadsheet used to create it. On a couple occasions, both are referred to by the same thing (the Research Plans and Goals chart). The Action Items in the above image come just after updating resource 2.03 with link for the Research Plans and Goals chart on the Wiki (i.e. you're updating it and then immediately replacing it with the spreadsheet). So the action items in this image should say something like "resource 2.04" and "the Research Plans and Goals Chart _Spreadsheet_. Obviously there should also be a 2.04 resource: "HfLA Volunteers Research Plans and Goals Chart Spreadsheet".

<img width="523" alt="image" src="https://user-images.githubusercontent.com/122848208/231632155-043dfa7a-4929-4384-af30-6a4b9c787b6d.png">
- The markdown website no longer includes these options

**Comment by pandanista** [comment]
Good catch. Thank you for the feedback! I have incorporated them into the template. 

I would like to know why the day, and frequency need to be included. It might make more sense not to include the day and frequency due to 1). the inconsistency and lack of bandwidth based on past weekly survey not fulfilled, 2). not having to update the chart later, 3). lack of day/date information at the planning phase since the chart is provided to participants or participants' guardians during the consent phase and we might not know the exact day/date to conduct a study yet.

**Comment by AndrewSalvatore** [comment]
Based on today's conversation at the TWE all-team meeting, we're are going to keep the Day and Frequency columns for the Plans and Goals Chart because it is used to inform volunteers, interns, and parents of expectations. For RPs that don't explicitly state the information, we'll enter TBD for them.

**Comment by pandanista** [comment]
The script link now directs to research plan docs. We will update the script link with google form link once created.

**Comment by priyatalwar** [comment]
Product reviewed it with UX Research lead
